WalletWasabi/Blockchain/TransactionBuilding/BnB/SelectionStrategy.cs

```diff
@@ -68,11 +80,30 @@ public virtual long ProcessAction(NextAction action, long[] selection, int depth
 
 			newSum = oldSum - selection[depth];
 			selection[depth] = 0;
+
+			VerifyInputCostsSum(selection);
 		}
 
 		return newSum;
 	}
 
+	private void VerifyInputCostsSum(long[] selection)
+	{
+		long inputCostsSum = 0;
+		for (int i = 0; i < selection.Length; i++)
+		{
+			if (selection[i] > 0)
+			{
+				inputCostsSum += InputCosts[i];
+			}
+		}
+
+		if (CurrentInputCosts != inputCostsSum)
+		{
+			throw new Exception("...");
```

Comment:
```suggestion
			throw new Exception("Input costs are out of sync. Aborting.");
```

Reply:
That's a leftover too. I just wanted to validate it does calculations correctly.